packages/flutter_tools/lib/src/build_system/targets/ios.dart

```diff
@@ -509,12 +510,32 @@ abstract class IosAssetBundle extends Target {
         .copySync(dsymOutputBinary.path);
     }
 
+    final FlutterProject flutterProject = FlutterProject.fromDirectory(environment.projectDir);
+
+    bool isImpellerEnabled() {
+      final PlistParser plistParser = PlistParser(fileSystem: globals.fs, logger: environment.logger, processManager: globals.processManager);
+
+      if (!flutterProject.ios.infoPlist.existsSync()) {
+        return false;
+      }
+      final Map<String, Object> info = plistParser.parseFile(flutterProject.ios.infoPlist.path);
+
+      if (!info.containsKey('FLTEnableImpeller')) {
```

Comment:
Don't need this condition with the next condition, if they key is missing you'll get `null` back from the map

Reply:
Removed. I also made a cast below this less spicy.